src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -329,18 +329,25 @@ public void initializeFromEntity(@NotNull final EntityCitizen entity)
      */
     private String generateName(@NotNull final Random rand)
     {
-        final String firstName;
-        if (!female)
+        String citizenName;
+        if (female)
         {
-            firstName = String.format("%s %s. %s", getRandomElement(rand, Configurations.femaleFirstNames), getRandomLetter(rand),
+            citizenName = String.format("%s %s. %s", getRandomElement(rand, Configurations.femaleFirstNames), getRandomLetter(rand),
                     getRandomElement(rand, Configurations.lastNames));
         }
         else
         {
-            firstName = String.format("%s %s. %s", getRandomElement(rand, Configurations.maleFirstNames), getRandomLetter(rand),
+            citizenName = String.format("%s %s. %s", getRandomElement(rand, Configurations.maleFirstNames), getRandomLetter(rand),
                     getRandomElement(rand, Configurations.lastNames));
         }
-        return String.format("%s %s. %s", firstName, getRandomLetter(rand), getRandomElement(rand, Configurations.lastNames));
+        for (int i = 1; i <= this.getColony().getMaxCitizens(); i++)
+        {
+            if (this.getColony().getCitizen(i) != null && this.getColony().getCitizen(i).getName().equals(citizenName))
+            {
+                citizenName = generateName(rand);
```

Comment:
Nojo created that a whole while ago already.
That's the same like in 1.10 I just copied it from there to here because merged broke it